docs/runbook.md

```diff
@@ -13,103 +13,195 @@
+
 ### test_pass
 
+type: str, optional, default is empty
+
+Set the test pass name of this test run. This name combined with test project will be used to group test results in html.
+
 ### tags
 
+type: list of str, optional, default is empty
+
+Set the tags of the test run. This name combined with test project and test pass names will be used to group test results in html.
+
 ### parent
 
+Share the runbook for similar runs.
+
 #### path
 
-#### strategy
+type: str, optional, default is empty
 
-Not implemented
+Specify the path of the parent runbook.
 
 ### extension
 
+type: list of str, optional, default: empty
+
+Specify the path of the LISA module for loading test cases and extensions.
+
 ### variable
 
+Used to support variables in other fields.
+
 #### is_secret
 
-#### file
+type: boolean, optional, default is False.
 
-#### name
+If the variable is secret, the value of this variable will be removed from log and other output information.
 
-### value
+#### file
 
-### artifact
+type: list of str, optional, default: empty
+
+Specify path of other yml files which define variables.
 
 #### name
 
-#### locations
+type: str, optional, default is empty.
 
-##### type
+Variable name.
 
-##### path
+#### value
+
+type: str, optional, default is empty
 
+Value of the paired variable.
 
 ### notifier
 
+Receive messages during the test run and output them somewhere. The supported messages are derived from `MessageBase`, including `TestRunMessage`, `TestResultMessage`, etc.
+
 #### console
 
+This is the simplest notifier. It outputs messages to the console and file log and demonstrates how to implement notification procedures.
+
+##### log_level
+
+type: str, optional, default: DEBUG, values: DEBUG, INFO, WARNING...
+
+Set log level of notification messages.
+
 #### html
 
+Output test results in html format. It can be used for local development or as the body of an email.
+
+##### path
+
+type: str, optional, default: lisa.html
+
+Specify the output file name and path.
+
+##### auto_open
+
+type: bool, optional, default: False
+
+When set to True, the html will be opened in the browser after completion. Useful in local run.
 
 ### environment
 
-#### name
+List of environments.
+
+#### environments
+
+Define test run environment.
+
+##### name
 
-#### topology
+type: str, optional, default is empty
 
-#### nodes
+Specify the name of the environment.
 
-#### nodes_requirement
+##### topology
 
-##### type
+type: str, optional, default is "subnet"
+
+Specify the topology of the environment.
+
+##### nodes
+
+List of system under test.
+
+##### nodes_requirement
+
+Testing required environment, e.g. node_count (default is 1), core_count (default is 1), memory_mb (default is 512 MB), disk_count (default is 1) etc. The SUT can be created once the node requirement is met.
+
+###### type
+
+type: str, optional, default is "requirement"
 
 ### platform
 
+List of platform, default value is 'ready', we also support 'azure' platform.
```

Comment:
list all of them here.

Reply:
This document is for users, who use LISA. For developers, we can add content to extension.md or docstrings. Could you show an example, how the layout is like? It may need more time to enable it, would you like to create issues to track them also?

Reply:
I'm not going the docstrings site route just yet, but yeah--that's the ideal way. But even users want to know the yml syntax in detail enough to be able to write new ones, even with the classes that already come with the code.